src/trio/_core/_exceptions.py

```diff
@@ -67,11 +73,49 @@ class Cancelled(BaseException, metaclass=NoPublicConstructor):
 
     """
 
+    source: Literal["deadline", "nursery", "explicit", "unknown", "KeyboardInterrupt"]
+    # TODO: this should probably be a Task?
+    source_task: str | None = None
```

Comment:
What are the pros and cons?  Off the top of my head, better access to the other Task attributes vs keeping tasks alive for longer is worse for GC.